src/test/FuzzerImpl.cpp

```diff
@@ -0,0 +1,358 @@
+    qSet0.validators.push_back(v10NodeID);
+    qSet0.validators.push_back(v11NodeID);
+
+    mSimulation->addNode(v10SecretKey, qSet0);
+    mSimulation->addNode(v11SecretKey, qSet0);
+
+    mSimulation->addPendingConnection(v10SecretKey.getPublicKey(),
+                                      v11SecretKey.getPublicKey());
+
+    mSimulation->startAllNodes();
+
+    // crank until nodes are connected
+    mSimulation->crankUntil(
+        [&]() {
+            auto nodes = mSimulation->getNodes();
+            auto numberOfSimulationConnections =
+                nodes[ACCEPTOR_INDEX]
+                    ->getOverlayManager()
+                    .getAuthenticatedPeersCount() +
+                nodes[INITIATOR_INDEX]
+                    ->getOverlayManager()
+                    .getAuthenticatedPeersCount();
+            return numberOfSimulationConnections == 2;
+        },
+        std::chrono::milliseconds{500}, false);
+}
+
+void
+OverlayFuzzer::inject(XDRInputFileStream& in)
+{
+    // see note on TransactionFuzzer's tryRead above
+    auto tryRead = [&in](StellarMessage& m) {
+        try
+        {
+            return in.readOne(m);
+        }
+        catch (...)
+        {
+            m.type(HELLO); // we ignore HELLO messages
+            return true;
+        }
+    };
+
+    StellarMessage msg;
+    while (tryRead(msg))
+    {
+        if (isBadOverlayFuzzerInput(msg))
+            return;
+
+        auto nodeids = mSimulation->getNodeIDs();
+        auto loopbackPeerConnection = mSimulation->getLoopbackConnection(
+            nodeids[INITIATOR_INDEX], nodeids[ACCEPTOR_INDEX]);
+
+        auto initiator = loopbackPeerConnection->getInitiator();
+        auto acceptor = loopbackPeerConnection->getAcceptor();
+
+        initiator->getApp().getClock().postToCurrentCrank(
+            [initiator, msg]() { initiator->Peer::sendMessage(msg); });
+
+        mSimulation->crankForAtMost(std::chrono::milliseconds{500}, false);
+
+        // clear all queues and cancel all events
+        initiator->clearInAndOutQueues();
+        acceptor->clearInAndOutQueues();
+
+        while (initiator->getApp().getClock().cancelAllEvents())
+            ;
+        while (acceptor->getApp().getClock().cancelAllEvents())
+            ;
+    }
+}
+
+namespace FuzzUtils
+{
+#define FUZZER_INITIAL_CORPUS_MAX_OPERATIONS 5
+// currently there are 13 types of operations
+#define FUZZER_INITIAL_CORPUS_OPERATION_GEN_SIZE 13
+void
+generateTransactionFuzzerInput(std::string const& filename)
+{
+    XDROutputFileStream out;
+    out.open(filename);
+    autocheck::generator<Operation> gen;
+    xdr::xvector<Operation> ops;
+    auto numops =
+        autocheck::generator<uint>()(FUZZER_INITIAL_CORPUS_MAX_OPERATIONS);
+    for (int i = 0; i < numops; ++i)
+    {
+        Operation op = gen(FUZZER_INITIAL_CORPUS_OPERATION_GEN_SIZE);
+        while (isBadTransactionFuzzerInput(op))
+        {
+            op = gen(FUZZER_INITIAL_CORPUS_OPERATION_GEN_SIZE);
```

Comment:
same comment than for overlay: don't use `FUZZER_INITIAL_CORPUS_OPERATION_GEN_SIZE` here but some higher number